src/main/resources/org/jboss/xavier/analytics/rules/workload/inventory/Complexity.drl

```diff
@@ -0,0 +1,100 @@
+package org.jboss.xavier.analytics.rules.workload.inventory;
+
+import org.jboss.xavier.analytics.pojo.input.workload.inventory.VMWorkloadInventoryModel
+import org.jboss.xavier.analytics.pojo.output.workload.inventory.WorkloadInventoryReportModel
+import java.util.HashSet;
+
+import function org.jboss.xavier.analytics.functions.HelperFunctions.isSupportedOS;
+import function org.jboss.xavier.analytics.functions.HelperFunctions.isUnsupportedOS;
+import function org.jboss.xavier.analytics.functions.HelperFunctions.isUndetectedOS;
+
+dialect "java"
+agenda-group "Complexity"
+lock-on-active true
+auto-focus false
+
+rule "No_Flag_Supported_OS"
+    when
+        workloadInventoryReport : WorkloadInventoryReportModel(
+            flagsIMS == null,
+            osName != null,
```

Comment:
```suggestion
            osDescription != null,
```

Reply:
Why are we matching on the osDescription rather than the osName? It seems to me that matching on the description will require far more changes as versions change, ie the differenec between matching on 'RHEL' (which would stay the same between versions) and 'Red Hat Enterprise Linux v7.6' (which wouldn't).

Reply:
I think I'm answering my own questions here :)

Is this why we are doing a contains check on 'Red Hat Enterprise Linux'?

Reply:
Exactly, you got the point.
The `osName` in the sample payload is always `Linux` that would mean having always `Unknown` complexity even if the OS, in many case, it's `Red Hat Enterprise Linux` which is not unknown :wink:

Reply:
Ah - so the osName doesn't distinguish between Linuxes? i didn't realise that.